x-pack/metricbeat/tests/integration/setup_integration_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package integration
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/beats/v7/libbeat/tests/integration"
+	"github.com/elastic/beats/v7/libbeat/version"
+)
+
+func TestIndexTotalFieldsLimitNotReached(t *testing.T) {
+	cfg := `
+metricbeat:
+logging:
+  level: debug
+metricbeat.config.modules:
+  path: ${path.config}/modules.d/*.yml
+  reload.enabled: false
+`
+	metricbeat := integration.NewBeat(t, "metricbeat", "../../metricbeat.test")
+	metricbeat.WriteConfigFile(cfg)
+	esURL := integration.GetESURL(t, "http")
+	kURL, _ := integration.GetKibana(t)
+
+	ver, _, _ := strings.Cut(version.GetDefaultVersion(), "-")
+	index := "metricbeat-" + ver
+
+	dataStreamURL, err := integration.FormatDatastreamURL(t, esURL, index)
+	require.NoError(t, err)
+	templateURL, err := integration.FormatIndexTemplateURL(t, esURL, index)
+	require.NoError(t, err)
+	policyURL, err := integration.FormatPolicyURL(t, esURL, index)
+	cleanUpES := func() {
+		_, _, err := integration.HttpDo(t, http.MethodDelete, dataStreamURL)
+		require.NoErrorf(t, err, "cleanup failed: could not remove datastream %s", index)
+		_, _, err = integration.HttpDo(t, http.MethodDelete, templateURL)
+		require.NoErrorf(t, err, "cleanup failed: could not remove index template %s", index)
+		_, _, err = integration.HttpDo(t, http.MethodDelete, policyURL)
+		require.NoErrorf(t, err, "cleanup failed: could not ilm policy %s", index)
+	}
+	// ensure datastream/index template/ ilm policy is set before running the test
+	cleanUpES()
+	t.Cleanup(cleanUpES)
+
+	metricbeat.Start("setup",
+		"--index-management",
+		"-E", "setup.kibana.protocol=http",
+		"-E", "setup.kibana.host="+kURL.Hostname(),
+		"-E", "setup.kibana.port="+kURL.Port(),
+		"-E", "output.elasticsearch.protocol=http",
+		"-E", "output.elasticsearch.hosts=['"+esURL.String()+"']",
+		"-E", "output.file.enabled=false")
+	procState, err := metricbeat.Process.Wait()
+	require.NoError(t, err, "metricbeat setup failed")
+	require.Equal(t, 0, procState.ExitCode(), "metricbeat setup failed: incorrect exit code")
+
+	// generate an event with dynamically mapped fields
+	fields := map[string]string{}
+	totalFields := 500
+	for i := range totalFields {
+		fields[fmt.Sprintf("a-label-%d", i)] = fmt.Sprintf("some-value-%d", i)
+	}
+	event, err := json.Marshal(map[string]any{
+		"@timestamp":        time.Now().Format(time.RFC3339),
+		"kubernetes.labels": fields,
```

Comment:
Why `kubernete.labels`? Wouldn't the same happen on any field? It it has to be `kubernetes.labels` it would be nice to add a comment explaining why.